src/ducks/paymentMethods.duck.js

```diff
@@ -1,2 +1,182 @@
+};
+
+export default function payoutMethodsPageReducer(state = initialState, action = {}) {
+  const { type, payload } = action;
+  switch (type) {
+    case CREATE_STRIPE_CUSTOMER_REQUEST:
+      return { ...state, createStripeCustomerError: null, createStripeCustomerInProgress: true };
+    case CREATE_STRIPE_CUSTOMER_SUCCESS:
+      return {
+        ...state,
+        createStripeCustomerInProgress: false,
+        stripeCustomer: payload,
+      };
+    case CREATE_STRIPE_CUSTOMER_ERROR:
+      console.error(payload);
+      return {
+        ...state,
+        createStripeCustomerError: payload,
+        createStripeCustomerInProgress: false,
+      };
+    case ADD_PAYMENT_METHOD_REQUEST:
+      return { ...state, addPaymentMethodError: null, addPaymentMethodInProgress: true };
+    case ADD_PAYMENT_METHOD_SUCCESS:
+      return {
+        ...state,
+        addPaymentMethodInProgress: false,
+        stripeCustomer: payload,
+      };
+    case ADD_PAYMENT_METHOD_ERROR:
+      console.error(payload);
+      return {
+        ...state,
+        addPaymentMethodError: payload,
+        addPaymentMethodInProgress: false,
+      };
+
+    case DELETE_PAYMENT_METHOD_REQUEST:
+      return { ...state, deletePaymentMethodError: null, deletePaymentMethodInProgress: true };
+    case DELETE_PAYMENT_METHOD_SUCCESS:
+      return {
+        ...state,
+        deletePaymentMethodInProgress: false,
+      };
+    case DELETE_PAYMENT_METHOD_ERROR:
+      console.error(payload);
+      return {
+        ...state,
+        deletePaymentMethodError: payload,
+        deletePaymentMethodInProgress: false,
+      };
+    default:
+      return state;
+  }
+}
+
+// ================ Action creators ================ //
+
+export const stripeCustomerCreateRequest = () => ({ type: CREATE_STRIPE_CUSTOMER_REQUEST });
+
+export const stripeCustomerCreateSuccess = stripeCustomer => ({
+  type: CREATE_STRIPE_CUSTOMER_SUCCESS,
+  payload: stripeCustomer,
+});
+
+export const stripeCustomerCreateError = e => ({
+  type: CREATE_STRIPE_CUSTOMER_ERROR,
+  payload: e,
+  error: true,
+});
+
+export const addPaymentMethodRequest = () => ({ type: ADD_PAYMENT_METHOD_REQUEST });
+
+export const addPaymentMethodSuccess = stripeCustomer => ({
+  type: ADD_PAYMENT_METHOD_SUCCESS,
+  payload: stripeCustomer,
+});
+
+export const addPaymentMethodError = e => ({
+  type: ADD_PAYMENT_METHOD_ERROR,
+  payload: e,
+  error: true,
+});
+
+export const deletePaymentMethodRequest = () => ({ type: DELETE_PAYMENT_METHOD_REQUEST });
+
+export const deletePaymentMethodSuccess = stripeCustomer => ({
+  type: DELETE_PAYMENT_METHOD_SUCCESS,
+  payload: stripeCustomer,
+});
+
+export const deletePaymentMethodError = e => ({
+  type: DELETE_PAYMENT_METHOD_ERROR,
+  payload: e,
+  error: true,
+});
+
+// ================ Thunks ================ //
+
+export const createStripeCustomer = params => (dispatch, getState, sdk) => {
+  const stripePaymentMethodId = params.setupIntent.payment_method;
```

Comment:
This is a bit odd. Why not just pass setupIntent instead of wrapping it with some object?

```js
export const createStripeCustomer = setupIntent => (dispatch, getState, sdk) => {
  const stripePaymentMethodId = setupIntent.payment_method;
```
or even paymentMethod directly:

```js
export const createStripeCustomer = paymentMethod => (dispatch, getState, sdk) => {
  const stripePaymentMethodId = paymentMethod;
```

i.e. extract the relevant data before calling these thunk functions.

Reply:
This is a bit related to the other branch on top of this one.